docs/proposals/rewrite.md

```diff
@@ -0,0 +1,71 @@
+# Rewrite the Agent
+
+- Authors: massi
+- Date: 2019-04-01
+- Status: draft
+- [Discussion](https://github.com/DataDog/datadog-agent/pull/0)
+
+## Overview
+
+Agent 6 was a complete rewrite of 5 using a different programming language and
+a different software architecture. It was fun to do, let's do it again.
+
+## Problem
+
+The real problem will be finding approval and resources for the project but let's
+not think about this for a moment.
+
+## Constraints
+
+1. Solution must be packaged with Omnibus Software
```

Comment:
It's great to see this in the requirements. that's definitely the #1 requirement for a smooth developer experience 💯